### ISSUE TITLE
Secure grafana-image-renderer

### ISSUE THREAD

**Issue by logiserv** [grafana/grafana-image-renderer#133]
**What would you like to be added**: Secure http access to grafana-image-render docker

**Why is this needed**: I run grafana-image-renderer successfully externally to grafana
I run it in a docker by Google Cloud Run.
I want to secure access to docker http output .
Can i add user/password or IP/DNS  sources to access grafana-image-renderer ?
How to modify rendering section of grafana.ini if i can add user/password access ?

**Comment by marefr** [comment]
I haven't tried, but think you should be able to do this:

```ini
[rendering]
server_url = http://<user>:<password>@<host>:<port>/render
```

**Comment by logiserv** [comment]
Only Basic Authentication is possible ? Basic Authentication isn't deprecated ?
A argument (with user/password) is possible when we launch grafana-image-renderer docker ?

**Comment by marefr** [comment]
Yeah currently. To validate basic auth you still have to have a reverse proxy in front, the image renderer doesn't support validating basic auth.

**Comment by marefr** [comment]
Would a secret sent as HTTP header from Grafana to Image Renderer service and validated solve your problem? Something I've thinking about.

**Comment by logiserv** [comment]
This is what i want (Google Sign-in section) :https://cloud.google.com/run/docs/authenticating/end-users#using_google_sign-in
https://developers.google.com/identity/sign-in/web/
Include the ID token in an Authorization: **Bearer** ID_TOKEN header in the request to the service

**Comment by logiserv** [comment]
@marefr sorry for my english ... not what i want but what you would make please.
Perhaps arguments for the docker user=$User and password=$Password can be a first step ? And User/Password in Rendering section of grafana.ini ?
Second step with few others authentications (at least Bearer Authentication) in docker's argument and Rendering section of grafana.ini ?

**Comment by pmenglund** [comment]
FYI including user and password in the `server_url` does not work, you still get redirected to `/login`.

**Comment by elliotmr** [comment]
> Would a secret sent as HTTP header from Grafana to Image Renderer service and validated solve your problem? Something I've thinking about.

Sorry about resurrecting an old thread, but this is exactly what I need. I can generate an Bearer token from the reverse proxy in front of the Grafana instance that will give the renderer service callback permissions.

**Comment by Proximyst** [comment]
Closing; this is what the renderer token is :)
